File: server/src/main/scala/kpn/api/common/route/RouteChangesPage.scala
package kpn.api.common.route

import kpn.api.common.changes.filter.ChangesFilterOption

case class RouteChangesPage(
  routeNameInfo: RouteNameInfo,
  filterOptions: Seq[ChangesFilterOption],
  changes: Seq[RouteChangeInfo],
  totalCount: Long,
  changeCount: Long
)

File: backend/api/src/main/java/kpn/api/common/route/RouteChangesPage.java
package kpn.api.common.route;

import kpn.api.common.changes.filter.ChangesFilterOption;
import kpn.api.common.route.RouteChangeInfo;
import kpn.api.common.route.RouteInfo;

import com.google.common.collect.ImmutableList;

public record RouteChangesPage(
  RouteInfo routeInfo,
  ImmutableList<ChangesFilterOption> filterOptions,
  ImmutableList<RouteChangeInfo> changes
) {
}
